src/test/java/ojosama/talkak/member/service/MemberServiceTest.java

```diff
@@ -0,0 +1,201 @@
+import ojosama.talkak.member.repository.MemberRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.transaction.annotation.Transactional;
+
+@SpringBootTest
+@Transactional
+class MemberServiceTest {
+
+    @Autowired
+    private MemberService memberService;
+    @Autowired
+    private MemberRepository memberRepository;
+    @Autowired
+    private MemberCategoryRepository memberCategoryRepository;
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+    Member member;
+    List<Category> categories;
+    List<CategoryType> categoriesNames = Arrays.asList(CategoryType.values());
+
+    /***
+     * 초기 회원 정보
+     * 성별: 남자
+     * 나이: 10대
+     * 선택 카테고리: 음식, 여행, 게임
+     */
+
+    @BeforeEach
+    void before() {
+        member = demoMember();
+        memberRepository.save(member);
+
+        List<Category> temp = categoriesNames.stream()
+            .map(cn -> new Category(null, cn))
+            .toList();
+        categories = categoryRepository.saveAll(temp);
+        System.out.println("출력: " + CategoryType.valueOf("FOOD"));
+        List<MemberCategory> memberCategories = categories.stream()
+            .map(c -> new MemberCategory(member, c))
+            .limit(3)
+            .toList();
+
+        memberCategoryRepository.saveAll(memberCategories);
+    }
+
+    @DisplayName("마이페이지 개인정보 가져오기")
+    @Test
+    void get_myPage_info() {
+        MyPageInfoResponse memberInfo = memberService.getMemberInfo(member.getId());
+
+        assertThat(memberInfo.gender()).isEqualTo("남자");
+        assertThat(memberInfo.age()).isEqualTo("20대");
+        assertThat(memberInfo.categories().size()).isEqualTo(3);
+    }
+
+    /***
+     * 회원 변경 정보
+     * 성별: 남자 -> 여자
+     * 나이: 10대 -> 20대
+     * 선택 카테고리: 음식, 여행, 게임 -> 음식, 음악, 스포츠
+     */
+
+    @DisplayName("마이페이지 개인정보 수정하기-성공")
+    @Test
+    void update_myPage_info() {
+        MyPageInfoRequest request = new MyPageInfoRequest("여자", "20대",
+            demoCategoryIds(Arrays.asList("음식", "음악", "스포츠")));
+        MyPageInfoResponse memberInfo = memberService.updateMemberInfo(member.getId(), request);
+        List<String> categoryNames = memberInfo.categories().stream()
+            .map(CategoryResponse::name)
+            .toList();
+
+        assertThat(memberInfo.gender()).isEqualTo("여자");
+        assertThat(memberInfo.age()).isEqualTo("20대");
+        assertThat(categoryNames.size()).isEqualTo(3);
+        assertThat(categoryNames).containsOnly("음식", "음악", "스포츠");
+    }
+
+    @DisplayName("마이페이지 개인정보 수정하기 실패-유효하지 않은 성별 정보")
+    @Test
+    void invalid_input_gender() {
+        assertErrorCode(MemberError.ERROR_UPDATE_MEMBER_INFO,
+            () -> memberService.updateMemberInfo(member.getId(), createGender(null)));
```

Comment:
테스트 코드 잘 작성하셨습니다 👍 💯 

memberService.updateMemberInfo() 가 중복으로 많이 호출 되고 있어요. 
서비스 운영을 하다보면 memberService.updateMemberInfo() 가 어디서 사용되는지 파악이 필요한 경우가 있어요. 그런데 테스트 코드에서 이렇게 직접 호출을 하게 되면 아래와 같이 호출자가 많이 나옵니다.

![image](https://github.com/user-attachments/assets/556a1238-a453-40f4-855c-78ae0c7e0839)

별도 메서드로 분리해보면 어떨까요?

```suggestion
        assertErrorCode(MemberError.ERROR_UPDATE_MEMBER_INFO,
            () -> updateMemberInfo(member.getId(), createGender(null)));
```